### ISSUE TITLE
Issue with DIISCO predicted interactions (W)

### ISSUE THREAD

**Issue by Trip-F** [azizilab/DIISCO_public#17]
Hello,

I am interested in applying your method to my own data and am first working with the CAR T cell coculture dataset you used in the tutorial to make sure i can reproduce your results with your data and code. I am following the DIISCO_tutorial.ipynb notebook. I am new to machine learning and only a little familar with python so please bear with me.

I successfully reproduced the predicted cell type proportions, which I believe comes from the tensor `f_samples_predict`. However, predicting interactions has not worked. 

Here is my code to generate the model. My hyperparameter values are the same as yours as is all other code to get to this point.

```{}
model = DIISCO(lambda_matrix=prior_matrix, hypers_init_vals=hyperparams, verbose=True)
model.fit(timepoints, 
          proportions, 
          n_iter=10000, 
          lr=0.00005,
          hypers_to_optim=[], 
          guide="MultivariateNormalFactorized")
```

Oddly, this produces a warning saying: 

`UserWarning: Found non-auxiliary vars in guide but not model, consider marking these infer={'is_auxiliary': True}: {'lengthscale_w'}`

My understanding is that this means that `MultivariateNormalFactorized` expects to optimize `LENGTHSCALE_W` but we don't tell the model to optimize this param so we get a warning. I think this has something to do with the model not correctly predicting interactions but I am not sure.

When I look at model loss, I notice that my convergence seems much less jumpy than yours and the initial losses are higher. See below. I suspect this is related to my issue but am not sure. allowing the model to optimize `LENGTHSCALE_W` did not fix this.

![Image](https://github.com/user-attachments/assets/a7147cad-ace3-4d69-935a-a95601618219)

So now when I extract my W and F tensors from 

`samples = model.sample(predict_timepoints, n_samples= 10000, include_emission_variance=True)`

I see that my predicted cell type abundance is just as you found. See below.

![Image](https://github.com/user-attachments/assets/5090d33a-e6c4-46ff-8385-958965ae7ee9)

But when I look at the interaction matrix averaged across time (W), i see a different result from yours, where it looks like all interactions are negative. This is not what you saw as you had multiple positive interactions.

![Image](https://github.com/user-attachments/assets/bd0a42e4-0290-494f-aa5d-a6333e72bfe2)

And likewise looking at interactions over time.

![Image](https://github.com/user-attachments/assets/0f607dd6-d932-49de-964f-49a8ce3e4996)

I am not getting any errors. Just that one warning above. I resolved that warning by changing the `hypers_to_optimize` variable in `model.fit` to `names.LENGTHSCALE_W`. This resolved the warning but did not make the results any better. The modified code chunk is:

```
model = DIISCO(lambda_matrix=prior_matrix, hypers_init_vals=hyperparams, verbose=True)
model.fit(timepoints, 
          proportions, 
          n_iter=10000, 
          lr=0.00005,
          hypers_to_optim=[names.LENGTHSCALE_W], 
          guide="MultivariateNormalFactorized")
```

The only other thing I have changed from the tutorial is something that has already been addressed here which is that model.sample gives a dict with each tensor so you have to retrieve the tensors with:

```
W_samples_predict = samples["W"]
f_samples_predict = samples["F"]
y_samples_predict = samples["Y"]
```
This fixed the error previously described but not the issue I am having.

Do you have any sense of what might be going wrong with my code? Am I on the right track or is something else going on? Apologies for the long post. I have never posted here so I hope that I am providing enough info. Please let me know if anything is unclear.

Edit: Not sure if this is useful or relevant but some potentially important package versions Im using:

Python: 3.10.18 (main, Jun  5 2025, 08:37:47) [Clang 14.0.6 ]
Torch: 2.8.0
Numpy: 2.2.4
diisco: 1.0
